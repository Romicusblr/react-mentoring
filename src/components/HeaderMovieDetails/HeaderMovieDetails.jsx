import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Logo from '../Logo';
import style from './HeaderMovieDetails.module.css';
import MovieDetails from '../MovieDetails';

const HeaderMovieDetails = ({ movie, className, ...other }) => (
  <header
    className={classNames(style.headerMovieDetails, className)}
    {...other}
  >
    <Logo />
    <MovieDetails movie={movie} />
  </header>
);

HeaderMovieDetails.propTypes = {
  movie: PropTypes.objectOf(),
  className: PropTypes.string,
};

HeaderMovieDetails.defaultProps = {
  movie: {},
  className: '',
};

export default HeaderMovieDetails;
