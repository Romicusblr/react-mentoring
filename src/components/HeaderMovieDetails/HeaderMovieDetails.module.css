.headerMovieDetails {

  position: relative;
  padding: 5rem var(--padding-header);
  display: flex;
  flex-direction: column;
  background-image: url('./bg.png');

  &>h3 {
    margin: 0;
    position: absolute;
    top: 1.5rem;
    left: 4rem;
  }
}
