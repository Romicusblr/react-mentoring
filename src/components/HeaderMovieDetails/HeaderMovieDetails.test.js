import HeaderMovieDetails from '.';

fdescribe('HeaderMovieDetails', () => {
  it('should be selectable by class "headerMovieDetails"', () => {
    expect(shallow(<HeaderMovieDetails />).is('.headerMovieDetails')).toBe(true);
  });

  it('should match to snapshot', () => {
    const snap = shallow(<HeaderMovieDetails />);
    expect(snap).toMatchSnapshot();
  });
});
