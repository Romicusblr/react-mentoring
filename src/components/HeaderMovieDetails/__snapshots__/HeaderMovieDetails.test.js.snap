// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderMovieDetails should match to snapshot 1`] = `
<header
  className="headerMovieDetails"
>
  <Logo
    className=""
  />
  <MovieDetails
    className=""
    movie={
      Object {
        "budget": 125000000,
        "genres": Array [
          "Adventure",
          "Action",
          "Crime",
          "Mystery",
        ],
        "id": 58574,
        "overview": "There is a new criminal mastermind at large (Professor Moriarty) and not only is he Holmesâ€™ intellectual equal, but his capacity for evil and lack of conscience may give him an advantage over the  detective.",
        "poster_path": "https://image.tmdb.org/t/p/w500/7OFqsVuS6DlVM10GUD72vCGpQm9.jpg",
        "release_date": "2011-11-22",
        "revenue": 334615000,
        "runtime": 129,
        "tagline": "The game is afoot.",
        "title": "Sherlock Holmes: A Game of Shadows",
        "vote_average": 7,
        "vote_count": 4823,
      }
    }
  />
</header>
`;
